### PR TITLE
Integrate similarity in server setup

### PR DESCRIPTION
**Pull request by mikkelkai**
# TODO
- [x] Create todo
- [x] Get cron job to recognize modules
- [x] Change cron job to every day instead of every minute
- [x] Change similarity docker compose to use Dockerfile-ss

### CODE REVIEW

File: src/similarity/similarity.py [hunk 1]
```diff
@@ -46,26 +59,33 @@ def _load_songs(songs):
...
 
-def _load_song(song_id, filename, segments):
+
+def _load_song(song_id, filename, segments, force = False):
     """ Loads song features from the database if
     available otherwise loading the file and
     loading features from it directly
     """
+
+    print('Loading: ' + song_id)
```
Comment (Casperfrc) [inline src/similarity/similarity.py]: Still needed in production?

File: src/similarity/similarity.py [hunk 2]
```diff
@@ -240,12 +260,24 @@ def _find_matches(searchContext):
     """ Searches for segments that
     are similar in the bucket
     """
+
     
     searchSegment, query_object = searchContext
+    print(searchSegment[1] + " - " + str(searchSegment[2]))
```
Comment (Casperfrc) [inline src/similarity/similarity.py]: Still needed in production?

File: src/similarity/similarity.py [hunk 3]
```diff
@@ -269,13 +301,19 @@ def analyze_songs(songs):
...
+
+def analyze_segments(segs):
     ss = SongSegment()
 
     count = ss.count()
 
     allMatches = list(map(lambda x: [], segs))
 
+    matchers = [Matcher.start().proxy() for _ in range(cpu_count())]
+    print("Searching through " + str(count // BUCKET_SIZE + 1) + " buckets, with " + str(BUCKET_SIZE) + " segments in each")
```
Comment (Casperfrc) [inline src/similarity/similarity.py]: Still needed in production?

File: src/similarity/similarity.py [hunk 4]
```diff
@@ -269,13 +301,19 @@ def analyze_songs(songs):
...
     for i in range(0, count // BUCKET_SIZE + 1):
+        print("Bucket: " + str(i + 1))
```
Comment (Casperfrc) [inline src/similarity/similarity.py]: Still needed in production?

File: src/similarity/similarity.py [hunk 5]
```diff
@@ -334,3 +375,23 @@ def analyze_songs(songs):
...
+                break
+
+        feature = _create_feature(np.frombuffer(segment['mfcc']), np.frombuffer(
+                segment['chroma']), np.frombuffer(segment['tempogram']))
+
+        segment_data.append((segment['_id'], segment['song_id'], segment['time_from'], feature))
+
+    s.close()
+
+    print("Updating similar for " + str(len(segment_data)) + " segments")
```
Comment (Casperfrc) [inline src/similarity/similarity.py]: Still needed in production?

File: src/similarity/similarity.py [hunk 6]
```diff
@@ -291,20 +329,23 @@ def analyze_songs(songs):
...
+        for i, seg in enumerate(segs):
+            matched.append(matchers[i % len(matchers)].match(seg, query_object))
 
-        matches = list(map(_find_matches, list(
-            map(lambda seg: (seg, query_object), segs))))
+        matches = pykka.get_all(matched)
 
-        for j in range(0, len(matches)):
+        """matches = list(map(_find_matches, list(
+            map(lambda seg: (seg, query_object), segs))))"""
```
Comment (Casperfrc) [inline src/similarity/similarity.py]: What is this?

File: src/__main__.py [hunk 7]
```diff
@@ -42,6 +43,8 @@ def main(argv):
     elif type == "video":
         v_app.run(host=cfg['rest_api_host_url'],
                   port=cfg['rest_api_host_port'], debug=cfg['rest_api_debug'])
+    elif type == "similarity":
+        print('similarity')
```
Comment (Casperfrc) [inline src/__main__.py]: Should this be doing something more?

File: src/similarity/similarity.py (hunk 1 above)
Comment (mikkelkai) [reply src/similarity/similarity.py]: Yes

File: src/similarity/similarity.py (hunk 2 above)
Comment (mikkelkai) [reply src/similarity/similarity.py]: Yes

File: src/similarity/similarity.py (hunk 3 above)
Comment (mikkelkai) [reply src/similarity/similarity.py]: Yes

File: src/similarity/similarity.py (hunk 4 above)
Comment (mikkelkai) [reply src/similarity/similarity.py]: Yes

File: src/similarity/similarity.py (hunk 5 above)
Comment (mikkelkai) [reply src/similarity/similarity.py]: Yes

File: src/similarity/similarity.py (hunk 6 above)
Comment (mikkelkai) [reply src/similarity/similarity.py]: Oops

File: src/__main__.py (hunk 7 above)
Comment (mikkelkai) [reply src/__main__.py]: You are reviewing a WIP pull ;)

File: Dockerfile-ss [hunk 8]
```diff
@@ -0,0 +1,23 @@
+FROM python:3.6
+
+RUN apt update && apt update
```
Comment (Casperfrc) [inline Dockerfile-ss]: halul

File: src/similarity/similarity.py [hunk 9]
```diff
@@ -334,3 +372,23 @@ def analyze_songs(songs):
...
     ss.close()
 
+
+def analyze_missing_similar():
+    s = SongSegment()
+
+    segment_data = []
+
+    for segment in s._db._db[s._dbcol].find({'similar.' + str(MATCHES - 1): {'$exists': False}}):
+        if segment['mfcc'] == None or segment['chroma'] == None or segment['tempogram'] == None:
```
Comment (Casperfrc) [inline src/similarity/similarity.py]: Lines too long

File: src/similarity/similarity.py [hunk 10]
```diff
@@ -334,3 +372,23 @@ def analyze_songs(songs):
...
+                break
+
+        feature = _create_feature(np.frombuffer(segment['mfcc']), np.frombuffer(
+                segment['chroma']), np.frombuffer(segment['tempogram']))
+
+        segment_data.append((segment['_id'], segment['song_id'], segment['time_from'], feature))
```
Comment (Casperfrc) [inline src/similarity/similarity.py]: Line too long

Comment (mikkelkai) [reply src/similarity/similarity.py]: fixed

File: src/similarity/similarity.py (hunk 9 above)
Comment (mikkelkai) [reply src/similarity/similarity.py]: autoformatter var uenig

**Pull request merged**